paddle/fluid/framework/transfer_scope_cache.cc

```diff
@@ -18,42 +18,62 @@ namespace paddle {
 namespace framework {
 
 #ifdef PADDLE_WITH_MKLDNN
-static std::unordered_map<size_t, Scope*>* static_transfer_data_cache = nullptr;
-static std::unordered_set<Scope*>* static_transfer_scope_cache = nullptr;
+using transfer_data_cache_map = std::unordered_map<size_t, Scope*>;
+using transfer_scope_cache_map = std::unordered_set<Scope*>;
+static std::unordered_map<size_t, transfer_data_cache_map*>
+    static_transfer_data_caches;
+static std::unordered_map<size_t, transfer_scope_cache_map*>
+    static_transfer_scope_caches;
 #endif
 
 std::unordered_map<size_t, Scope*>& global_transfer_data_cache() {
 #ifdef PADDLE_WITH_MKLDNN
-  // if get_cur_thread_id() == -1, means not using thread local method to do
-  // cache
-  if (platform::get_cur_thread_id() == -1) {
-    if (!static_transfer_data_cache)
-      static_transfer_data_cache = new std::unordered_map<size_t, Scope*>;
-    return *static_transfer_data_cache;
-  } else {
-#endif
-    thread_local auto* x = new std::unordered_map<size_t, Scope*>;
-    return *x;
-#ifdef PADDLE_WITH_MKLDNN
+  size_t tid = static_cast<size_t>(platform::get_cur_thread_id());
+
+  // if tid != 0, means there is specific mkldnn tid setting from user.
+  if (tid != 0) {
```

Comment:
How about use CONST value to describe "0" or "-1" mode, such as `constexpr char kEnableCacheRuntimeContext` etc. You can put these CONST value somewhere.

Reply:
change in PR #18453 .

Reply:
solved